src/SwaggerUiServiceProvider.php

```diff
@@ -36,12 +36,13 @@ public function register() : void
 
     protected function loadRoutes() : void
     {
-        Route::middleware(config('swagger-ui.middleware', ['web', EnsureUserIsAuthorized::class]))
-            ->prefix(config('swagger-ui.path'))
-            ->group(function () {
-                Route::view('/', 'swagger-ui::index')->name('swagger-ui');
-
-                Route::get('openapi.json', OpenApiJsonController::class)->name('swagger-openapi-json');
-            });
+        collect(config('swagger-ui.files'))->each(function ($values) {
+            Route::middleware($values['middleware'])
+                ->group(function () use ($values) {
+                    Route::get($values['path'], SwaggerViewController::class)->name($values['path'] . '.index');
+
+                    Route::get($values['path'] . '/{filename}', OpenApiJsonController::class)->name($values['path'] . '.json');
```

Comment:
why did you change it to not using a path variable for first part of the route? manually now having to get the first segment of the route in the controller seems not ideal in my opinion